src/batch.js

```diff
@@ -10,6 +11,49 @@ function mountOnto(router) {
   });
 }
 
+function parseURL(URL) {
+  if (typeof URL === 'string') {
+    return url.parse(URL)
+  }
+  return undefined;
+}
+
+function mBatchRoutingPath(originalUrl, serverURL, publicServerURL) {
+    serverURL = serverURL ? parseURL(serverURL) : undefined;
+    publicServerURL = publicServerURL ? parseURL(publicServerURL): undefined;
+
+    let apiPrefixLength = originalUrl.length - batchPath.length;
+    let apiPrefix = originalUrl.slice(0, apiPrefixLength);
+
+    let makeRoutablePath = function(requestPath) {
+      // The routablePath is the path minus the api prefix
+      if (requestPath.slice(0, apiPrefixLength) != apiPrefix) {
```

Comment:
minor: could replace apiPrefixLength with apiPrefix.length here and on line 35 and then remove line 44